it/controllers/benefits/reimbursed/VouchersOrCreditCardsBenefitsControllerISpec.scala

```diff
@@ -0,0 +1,534 @@
+import models.User
+import models.benefits.{BenefitsViewModel, ReimbursedCostsVouchersAndNonCashModel}
+import models.mongo.{EmploymentCYAModel, EmploymentDetails, EmploymentUserData}
+import org.jsoup.Jsoup
+import org.jsoup.nodes.Document
+import play.api.http.HeaderNames
+import play.api.http.Status.{BAD_REQUEST, OK, SEE_OTHER}
+import play.api.libs.ws.WSResponse
+import utils.{EmploymentDatabaseHelper, IntegrationTest, ViewHelpers}
+
+class VouchersOrCreditCardsBenefitsControllerISpec extends IntegrationTest with ViewHelpers with EmploymentDatabaseHelper {
+
+  val taxYearEOY: Int = taxYear - 1
+  val employmentId: String = "001"
+
+  private val userRequest = User(mtditid, None, nino, sessionId, affinityGroup)(fakeRequest)
+
+  private def employmentUserData(isPrior: Boolean, employmentCyaModel: EmploymentCYAModel): EmploymentUserData =
+    EmploymentUserData(sessionId, mtditid, nino, taxYearEOY, employmentId,
+      isPriorSubmission = isPrior, hasPriorBenefits = isPrior, employmentCyaModel)
+
+  def cyaModel(employerName: String, hmrc: Boolean, benefits: Option[BenefitsViewModel] = None): EmploymentCYAModel =
+    EmploymentCYAModel(EmploymentDetails(employerName, currentDataIsHmrcHeld = hmrc), benefits)
+
+  def benefits(reimbursedCostsVouchersAndNonCashModel: ReimbursedCostsVouchersAndNonCashModel): BenefitsViewModel =
+    BenefitsViewModel(carVanFuelModel = Some(fullCarVanFuelModel), accommodationRelocationModel = Some(fullAccommodationRelocationModel),
+      travelEntertainmentModel = Some(fullTravelOrEntertainmentModel), utilitiesAndServicesModel = Some(fullUtilitiesAndServicesModel),
+      isUsingCustomerData = true, isBenefitsReceived = true, medicalChildcareEducationModel = Some(fullMedicalChildcareEducationModel),
+      incomeTaxAndCostsModel = Some(fullIncomeTaxAndCostsModel),
+      reimbursedCostsVouchersAndNonCashModel = Some(reimbursedCostsVouchersAndNonCashModel))
+
+  private def pageUrl(taxYear: Int) = s"$appUrl/$taxYear/benefits/vouchers-or-credit-cards?employmentId=$employmentId"
+
+  val continueLink = s"/update-and-submit-income-tax-return/employment-income/$taxYearEOY/benefits/vouchers-or-credit-cards?employmentId=$employmentId"
+
+  object Selectors {
+    val captionSelector: String = "#main-content > div > div > form > div > fieldset > legend > header > p"
+    val paragraphSelector: String = "#main-content > div > div > form > div > fieldset > legend > p"
+    val continueButtonSelector: String = "#continue"
+    val continueButtonFormSelector: String = "#main-content > div > div > form"
+    val yesSelector = "#value"
+    val noSelector = "#value-no"
+  }
+
+  trait SpecificExpectedResults {
+    val expectedTitle: String
+    val expectedHeading: String
+    val paragraph: String
+    val expectedErrorTitle: String
+    val expectedErrorText: String
+  }
+
+  trait CommonExpectedResults {
+    val expectedCaption: String
+    val expectedButtonText: String
+    val yesText: String
+    val noText: String
```

Comment:
the bullet1 and bullet2 text need adding to this trait

Reply:
Updated as requested.